### PR TITLE
unison: update to 2.53.3

### PR DESCRIPTION
**Pull request by EbiArnie**
- Updates unison to 2.53.3
- Removes old versions - they did not build for me with OCaml 4.13.1 that is currently in Spack. Also, the changes from the previous versions seem to be quite significant.
- Used MakefilePackage and simplified package.py

### CODE REVIEW

File: var/spack/repos/builtin/packages/unison/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/unison/package.py:23]: Versions are typically deprecated for a release before being removed.  See the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions.

**Review by tldahlgren — COMMENTED** [review]
Confirmed the new version sha256 but suggest deprecating the older versions.  Will also defer to @hseara as the maintainer.

File: var/spack/repos/builtin/packages/unison/package.py
Comment (EbiArnie) [reply var/spack/repos/builtin/packages/unison/package.py:23]: I added the older versions back in, but I was unable to test this.
On develop, OCaml < 4.13.1 does not build for me.
And with OCaml 4.13.1, the older unison versions don't build.

I think the older versions are broken in develop without further patching to make them 4.13.1 compatible.

**Review by tldahlgren — APPROVED** [review]

**Comment by tldahlgren** [comment]
@hseara Please review this PR since you are the maintainer.

**Review by bernhardkaindl — APPROVED** [review]
@hseara now had more than 2 weeks to review the package and was again reminded 5 days ago.

Approving therefore as well with for the plan to merge.

As for generic recommendation for the removal of old versions:
> Versions are typically deprecated for a release before being removed. See the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions.

@EbiArnie: In this case I'd argue that it should not apply as we no longer have the easy possibility to test building these old versions. With referral to that, you may very well submit a new PR only for the removal of these then already deprecated old versions and add me as a reviewer. I'll see then if we can get them removed.

**Pull request merged**
